boards/README.mkdn

```diff
@@ -0,0 +1,8 @@
+# Board definitions
```

Comment:
Do we have a full list of "adding a new board" steps written down somewhere?

Reply:
I don't believe we do. Eventually I'd like "add a file here" to be the whole list, but as you may have noticed, xtask flash and friends still have hardcoded match statements after this change. (I would love to fix that.)